Libraries/StyleSheet/StyleSheet.js

```diff
@@ -93,9 +94,19 @@ module.exports = {
    */
   create(obj: {[key: string]: any}): {[key: string]: number} {
     var result = {};
+
     for (var key in obj) {
-      StyleSheetValidation.validateStyle(key, obj);
-      result[key] = StyleSheetRegistry.registerStyle(obj[key]);
+
+      let {ios, android, ...style} = obj[key];
+      if (ios && Platform.OS === 'ios') {
+        style = {...style, ...ios};
+      }
+      if (android && Platform.OS === 'android') {
+        style = {...style, ...android};
+      }
+
+      StyleSheetValidation.validateStyle(key, style);
```

Comment:
Hmmm... You're right. I'll take a look at this again. Thanks for pointing it out.